api/nodes/serializers.py

```diff
@@ -485,58 +485,88 @@ def get_absolute_url(self, obj):
             }
         )
 
+class InstitutionRelated(JSONAPIRelationshipSerializer):
+    id = ser.CharField(source='_id', required=False, allow_null=True)
+    class Meta:
+        type_ = 'institutions'
 
-class NodeInstitutionRelationshipSerializer(ser.Serializer):
-    id = ser.CharField(source='institution_id', required=False, allow_null=True)
-    type = TypeField(required=False, allow_null=True)
+class NodeInstitutionsRelationshipSerializer(ser.Serializer):
+    data = ser.ListField(child=InstitutionRelated())
+    links = LinksField({'self': 'get_self_url',
+                        'html': 'get_related_url'})
 
-    links = LinksField({
-        'self': 'get_self_link',
-        'related': 'get_related_link',
-    })
+    def get_self_url(self, obj):
+        return obj['self'].institutions_relationship_url()
+
+    def get_related_url(self, obj):
+        return obj['self'].institutions_url()
 
     class Meta:
         type_ = 'institutions'
 
-    def get_self_link(self, obj):
-        return obj.institution_relationship_url()
+    def get_institutions_to_add_remove(self, institutions, new_institutions):
+        diff = relationship_diff(
+            current_items={inst._id: inst for inst in institutions},
+            new_items={inst['_id']: inst for inst in new_institutions}
+        )
+
+        insts_to_add = []
+        for inst_id in diff['add']:
+            inst = Institution.load(inst_id)
+            if not inst:
+                raise exceptions.NotFound(detail='Institution with id "{}" was not found'.format(inst_id))
+            insts_to_add.append(inst)
+
+        return insts_to_add, diff['remove'].values()
 
-    def get_related_link(self, obj):
-        return obj.institution_url()
+    def make_instance_obj(self, obj):
+        return {
+            'data': obj.affiliated_institutions,
+            'self': obj
+        }
 
     def update(self, instance, validated_data):
-        node = instance
+        node = instance['self']
         user = self.context['request'].user
 
-        inst = validated_data.get('institution_id', None)
-        if inst:
-            inst = Institution.load(inst)
-            if not inst:
-                raise exceptions.NotFound
-            try:
-                node.add_primary_institution(inst=inst, user=user)
-            except UserNotAffiliatedError:
-                raise exceptions.ValidationError(detail='User not affiliated with institution')
-            node.save()
-            return node
-        node.remove_primary_institution(user)
+        add, remove = self.get_institutions_to_add_remove(
+            institutions=node.affiliated_institutions,
```

Comment:
Equivalent to `instance['data']`, consider normalizing with [this comment](https://github.com/CenterForOpenScience/osf.io/pull/5450#discussion_r62914890)